File: app/src/main/java/com/carpinelli/testeandroid/model/TypeField.java
package com.carpinelli.testeandroid.model;

public enum TypeField {
    TEXT(1), TELNUMBER(2), EMAIL(3);

    private final int id;

    TypeField(int id) {
        this.id = id;
    }

    public int getValue() {
        return id;
    }
}

File: app/src/main/java/com/santander/luizlago/testeandroid/enums/TypeField.kt
package com.santander.luizlago.testeandroid.enums

enum class TypeField(val value: String) {
    TEXT("1"),
    TELNUMBER("telnumber"),
    EMAIL("3");

    companion object {
        fun valueOf(value: String): TypeField? = TypeField.values().find { it.value == value }
    }
}

/**
 * Returns an enum entry with the specified name or `null` if no such entry was found.
 */
inline fun TypeFieldValueOfOrNull(name: String): com.santander.luizlago.testeandroid.enums.TypeField? {
    return com.santander.luizlago.testeandroid.enums.TypeField.values().find { it.value == name }
}